File: mbw/src/main/java/com/mycelium/wallet/activity/main/model/ActionButton.java
package com.mycelium.wallet.activity.main.model;

import java.util.Objects;

public class ActionButton {
    public int id;
    public int icon;
    public String text;
    public int textColor = 0;

    public ActionButton(int actionId, String text) {
        this(actionId, text, 0);
    }

    public ActionButton(int actionId, String text, int icon) {
        this.id = actionId;
        this.icon = icon;
        this.text = text;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ActionButton that = (ActionButton) o;
        return id == that.id &&
                icon == that.icon &&
                textColor == that.textColor &&
                Objects.equals(text, that.text);
    }

    @Override
    public int hashCode() {

        return Objects.hash(id, icon, text, textColor);
    }
}

File: mbw/src/main/java/com/mycelium/wallet/activity/main/model/ActionButton.kt
package com.mycelium.wallet.activity.main.model

import android.os.Bundle
import com.mycelium.wallet.activity.main.BuySellFragment
import java.util.*

class ActionButton @JvmOverloads constructor(val id: BuySellFragment.ACTION, var text: String,
                                             val icon: Int = 0, var iconUrl: String? = null,
                                             var args: Bundle = Bundle()) {
    var textColor = 0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as ActionButton?
        return id == that!!.id &&
                icon == that.icon &&
                textColor == that.textColor &&
                text == that.text
    }

    override fun hashCode(): Int {
        return Objects.hash(id, icon, text, textColor)
    }
}
